### PR TITLE
Display inbox count in the WP Dashboard

### PR DESCRIPTION
**Pull request by saifsultanc**
## Description
Added the red bubble inbox count display with the Workflow Label on the WP Dashboard

## Testing instructions
Switch to this branch and confirm the presence of the 'red bubble' with the inbox entries count next to the 'Workflow' on the WP Dashboard.

## Automated Test Enhancements
N/A
 
## Screenshots
N/A

## Documentation Changes?
N/A

## Checklist:
- [x] I've tested the code.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code follows the inline documentation standards.

### CODE REVIEW

File: class-gravity-flow.php [hunk 1]
```diff
@@ -218,6 +218,8 @@ public function init() {
 				)
 			);
 
+			add_filter('add_menu_classes', array( $this, 'show_notification_count' ), 10);
```
Comment (stevehenty) [inline class-gravity-flow.php:221]:
```suggestion
			add_filter( 'add_menu_classes', array( $this, 'show_notification_count' ), 10);
```

File: class-gravity-flow.php [hunk 2]
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
 			}
 		}
 
+		/**
+		 * Add inbox notification count to Workflow Menu.
+		 *
```
Comment (stevehenty) [inline class-gravity-flow.php:5964]: missing since tag

File: class-gravity-flow.php [hunk 3]
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
...
+		 * @param array $menu The current WP Dashboard Menu.
+		 */		
+		public function show_notification_count( $menu ) {
+			$workflow_menu_pos = -1;
+			foreach ( $GLOBALS['menu'] as $menuitem ) {
+				if( $menuitem[0] == "Workflow" ) {
```
Comment (stevehenty) [inline class-gravity-flow.php:5970]:
```suggestion
				if ( $menuitem[0] == "Workflow" ) {
```

File: class-gravity-flow.php [hunk 4]
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
...
 		}
 
+		/**
+		 * Add inbox notification count to Workflow Menu.
+		 *
+		 * @param array $menu The current WP Dashboard Menu.
+		 */		
+		public function show_notification_count( $menu ) {
+			$workflow_menu_pos = -1;
+			foreach ( $GLOBALS['menu'] as $menuitem ) {
```
Comment (stevehenty) [inline class-gravity-flow.php:5969]: Let's use `global $menu` at the top instead of using $GLOBALS.

File: class-gravity-flow.php [hunk 5]
```diff
@@ -5957,6 +5959,25 @@ public function filter_after_update_entry( $form, $entry_id ) {
...
+				if( $menuitem[0] == "Workflow" ) {
+					$workflow_menu_pos = array_search( $menuitem, $GLOBALS['menu'], true );
+				}
+			}
+
+			$pending_count = $this->get_workflow_count();
+			$menu[$workflow_menu_pos][0] = sprintf( __('Workflow %s'), "<span class='update-plugins count-$pending_count'><span class='plugin-count'>" . number_format_i18n($pending_count) . "</span></span>" );
```
Comment (stevehenty) [inline class-gravity-flow.php:5976]:
```suggestion
			$menu[ $workflow_menu_pos ][0] = sprintf( __( 'Workflow %s' ), "<span class='update-plugins count-$pending_count'><span class='plugin-count'>" . number_format_i18n($pending_count) . "</span></span>" );
```

Comment (stevehenty) [reply class-gravity-flow.php:5976]: Please check that this also works after changing the local of the site.

File: class-gravity-flow.php [hunk 6]
```diff
@@ -6506,6 +6527,23 @@ public function get_user_roles( $user_id = null ) {
...
+		/**
+		 * Return the inbox entries count from transient.
+		 * 
+		 * @since 2.5.12
+		 * 
+		 * @return int
+		 */		
+		public function get_workflow_count() {
+			$count_value = get_transient( 'gflow_inbox_count' );
+			if( $count_value === false) {
```
Comment (stevehenty) [inline class-gravity-flow.php:6539]:
```suggestion
			if ( $count_value === false) {
```

File: class-gravity-flow.php [hunk 7]
```diff
@@ -6506,6 +6527,23 @@ public function get_user_roles( $user_id = null ) {
...
+				$count_value = Gravity_Flow_API::get_inbox_entries_count();
+				set_transient( 'gflow_inbox_count', $count_value, MINUTE_IN_SECONDS );
```
Comment (stevehenty) [inline class-gravity-flow.php:6541]: This will impact all users on the site.

**Comment by saifsultanc** [comment]
> This is working but only if there's only one user on the site using the inbox. We'll need to handle multiple users. I'm also seeing some errors when the "Workflow" label is customised in the label settings.
> `PHP Notice: Undefined offset: 2 in /app/public/wp-admin/menu-header.php on line 91`

Fixed the custom label settings issue. Issues for multiple users also addressed, along with the formatting fixes.

File: change_log.txt [hunk 8]
```diff
@@ -1 +1,2 @@
 - Fixed an issue where GP Limit Dates does not function on the User Input step when the min and max range is based on non-editable Date fields.
+- Added the red bubble inbox count display on the WP Dashboard Workflow display.
```
Comment (stevehenty) [inline change_log.txt:2]:
```suggestion
- Added the red bubble inbox count display on the WP Dashboard Workflow menu item.
```

File: class-gravity-flow.php [hunk 9]
```diff
@@ -6506,6 +6534,23 @@ public function get_user_roles( $user_id = null ) {
...
 
+		/**
+		 * Return the inbox entries count from transient.
+		 * 
+		 * @since 2.5.12
+		 * 
+		 * @return int
+		 */		
+		public function get_workflow_count() {
+			$count_value = get_transient( 'gflow_inbox_count' . get_current_user_id()  );
```
Comment (stevehenty) [inline class-gravity-flow.php:6545]:
```suggestion
			$count_value = get_transient( 'gflow_inbox_count_' . get_current_user_id()  );
```

File: class-gravity-flow.php [hunk 10]
```diff
@@ -6506,6 +6534,23 @@ public function get_user_roles( $user_id = null ) {
...
 		}
 
+		/**
+		 * Return the inbox entries count from transient.
+		 * 
+		 * @since 2.5.12
+		 * 
+		 * @return int
+		 */		
+		public function get_workflow_count() {
```
Comment (stevehenty) [inline class-gravity-flow.php:6544]: Would this make more sense?
```suggestion
		public function get_inbox_count() {
```

File: class-gravity-flow.php [hunk 11]
```diff
@@ -5957,6 +5959,32 @@ public function filter_after_update_entry( $form, $entry_id ) {
...
 		}
 
+		/**
+		 * Add inbox notification count to Workflow Menu.
+		 *
+		 * @since 2.5.12
+		 * 
+		 * @param array $menu The current WP Dashboard Menu.
+		 */		
+		public function show_notification_count( $menu ) {
```
Comment (stevehenty) [inline class-gravity-flow.php:5969]: Would this be better?
```suggestion
		public function show_inbox_count( $menu ) {
```

**Pull request merged**